src/main/java/com/github/manolo8/darkbot/core/objects/facades/AssemblyMediator.java

```diff
@@ -0,0 +1,276 @@
+
+        //get list of shown recipe
+        long recipeCollectionAddress = Main.API.readMemoryPtr(address + 0x60);
+        recipesPtr.update(Main.API.readMemoryPtr(recipeCollectionAddress + 0x20));
+        recipesPtr.sync(recipes, Recipe::new);
+
+        //get selected recipe info
+        long selectedRecipeAddress = Main.API.readMemoryPtr(address + 0x70);
+        selectedRecipe.update(selectedRecipeAddress);
+
+        //get list of selected filters
+        long itemFilterViewController = Main.API.readMemoryPtr(address + 0x78);
+        rowSettingsArr.update(Main.API.readMemoryPtr(itemFilterViewController + 0xb0));
+        rowSettingsArr.sync(rowSettings, RowFilter::new);
+        filters.clear();
+        for (int i = 0; i < rowSettings.size(); i++) {
+            RowFilter currRow = rowSettings.get(i);
+            if (currRow.getFirst() != null) {
+                filters.add(new Filter(currRow.getFirst(), i, 0));
+            }
+            if (currRow.getSecond() != null) {
+                filters.add(new Filter(currRow.getSecond(), i, 1));
+            }
+        }
+
+        //get filter drop down is open
+        long filterDropdownAddress = Main.API.readMemoryPtr(itemFilterViewController + 0x60);
+        isFilterDropDownOpen = API.readBoolean(filterDropdownAddress + 0x1D0);
+    }
+
+    @Override
+    public int getSelectedRecipeIndex() {
+        return selectedRecipeIndex;
+    }
+
+    @Override
+    public AssemblyAPI.Recipe getSelectedRecipe() {
+        return selectedRecipe;
+    }
+
+    @Override
+    public boolean isFilterDropDownOpen() {
+        return isFilterDropDownOpen;
+    }
+
+    @Override
+    public List<? extends AssemblyAPI.Recipe> getRecipes() {
+        return recipes;
+    }
+
+    @Override
+    public List<? extends AssemblyAPI.RowFilter> getRowFilters() {
+        return rowSettings;
+    }
+
+    @Override
+    public List<? extends AssemblyAPI.Filter> getFilters() {
+        return filters;
+    }
+
+    public static class Recipe extends Auto implements AssemblyAPI.Recipe {
+        public String recipeId = "";
+        public List<String> rewards = new ArrayList<>();
+        public final ObjArray rewardsArr = ObjArray.ofVector(true);
+
+        public List<ResourceRequired> resourcesRequired = new ArrayList<>();
+        private final ObjArray resourcesRequiredArr = ObjArray.ofVector(true);
+
+        public boolean isCraftable = false;
+
+        @Override
+        public void update() {
+            if (address == 0) return;
+            isCraftable = API.readBoolean(address + 0x20);
+
+            long itemVoAddress = Main.API.readMemoryPtr(address + 0x58);
+            recipeId = API.readMemoryString(itemVoAddress, 0x48);
+
+            long rewardsArrAddress = Main.API.readMemoryPtr(address + 0x60);
+            rewardsArr.update(rewardsArrAddress);
+            rewards.clear();
+            rewardsArr.forEach(ptr -> rewards.add(API.readMemoryString(ptr, 0x48)));
```

Comment:
is it possible that this only needs to run on `update(long addr)`, that being, if the object is being read from a diff address? It would be nice for performance if stuff that does not ever update didn't need to update